lib/model/folder_summary.go

```diff
@@ -180,7 +178,7 @@ func (c *folderSummaryService) listenForUpdates(ctx context.Context) {
 		case ev := <-sub.C():
 			c.processUpdate(ev)
 		case <-ctx.Done():
-			return
+			return nil
```

Comment:
Is there a difference between returning nil or ctx.Error() in all these cases? (ctx.Error() would be less surprising to me)

Reply:
My thinking was that being cancelled isn't actually a failure but a shutdown as requested, and `err != nil` usually represents a failure. However I agree the existence of `context.Cancelled` implementing `error` challenges that assertion. Our service types have an `Error()` method which remember the error the function returns. I went looking for where we use that, and the only place I found is connections. So if the user e.g. disables relays they will get no error or a context cancelled error in the GUI next to the relay address. For dialers we do filter context cancelled errors:

```golang
func (s *connectionStatusHandler) setConnectionStatus(address string, err error) {
	if errors.Cause(err) == context.Canceled {
		return
	}
```

I'd go with returning `ctx.Error()` and expect anyone running a service to know that `context.Cancelled` is special. Respectively in most cases it won't be relevant anyway, because we are shutting down anyway.

Reply:
Right, I hade a vague feeling that it was possible to cancel a context with an arbitrary error, but it's not so the only possible error returns are context.Canceled and context.DeadlineExceeded. I think it would make sense to return the error regardless; in all cases I can imagine the context cancelling should mean we are shutting down so no action will be taken based on it... At least the `case err := <- ctx.Done: return err` is visually common and unconfusing